taichi/backends/cuda/codegen_cuda.cpp

```diff
@@ -425,19 +425,20 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
           auto val_type = ptr_type->get_pointee_type();
           llvm::Value *data_ptr = nullptr;
           llvm::Value *bit_offset = nullptr;
+          Type* int_in_mem = nullptr;
           if (auto cit = val_type->cast<CustomIntType>()) {
+            int_in_mem = val_type;
             dtype = cit->get_physical_type();
           } else if (auto cft = val_type->cast<CustomFloatType>()) {
-            dtype = cft->get_compute_type()
-                        ->as<CustomIntType>()
-                        ->get_physical_type();
+            int_in_mem = cft->get_digits_type();
+            dtype = int_in_mem->as<CustomIntType>()->get_physical_type();
```

Comment:
The `compute_type` of `CustomFloatType` is f32, f64,... which cannot be `CustomIntType`